sql/src/main/java/org/elasticsearch/action/bulk/SymbolBasedBulkShardProcessor.java

```diff
@@ -273,6 +302,64 @@ private void executeRequests() {
         }
     }
 
+    private void createPendingIndices() {
+        final List<PendingRequest> pendings = new ArrayList<>();
+        final Set<String> indices;
+
+        synchronized (requestsForNewIndices) {
+            indices = ImmutableSet.copyOf(Sets.difference(requestsForNewIndices.keySet(), indicesCreated));
+            for (Map.Entry<String, List<PendingRequest>> entry : requestsForNewIndices.entrySet()) {
+                pendings.addAll(entry.getValue());
+            }
+            requestsForNewIndices.clear();
+            pendingNewIndexRequests.set(0);
+        }
+
+        if (indices.size() > 0 && Iterables.any(indices, shouldAutocreateIndexPredicate)) {
```

Comment:
shouldn't you just use indices here where autoCreate is actually true? so either Iterables.all or use the filtered indices?